subsys/fs/fcb/fcb_elem_info.c

```diff
@@ -63,14 +62,49 @@ fcb_elem_crc8(struct fcb *fcb, struct fcb_entry *loc, uint8_t *c8p)
 	return 0;
 }
 
+int
+fcb_elem_endmarker(struct fcb *_fcb, struct fcb_entry *loc, uint8_t *c8p)
+{
+	if (loc->fe_elem_off + 2 > loc->fe_sector->fs_size) {
+		return -ENOTSUP;
+	}
+
+	if (IS_ENABLED(CONFIG_FCB_ALLOW_FIXED_ENDMARKER) &&
+	    (_fcb->f_flags & FCB_FLAGS_CRC_DISABLED)) {
+
+		uint8_t tmp_str[FCB_TMP_BUF_SZ];
```

Comment:
It should be possible to move lines 75 - 90 to the start of the routine and not to duplicate this in fcb_elem_crc8 (the tmp_str to calculate the len is only a 2 byte string).

Reply:
Ok, I've restructured the `fcb_elem_endmarker` and  `fcb_elem_crc8` to remove the `tmp_str` from `fcb_elem_crc8` and to remove the flag in` fcb` struct when `CONFIG_FCB_ALLOW_FIXED_ENDMARKER` is not enabled. Application size with `CONFIG_FCB_ALLOW_FIXED_ENDMARKER` disabled is now the same as without this PR.